NMS-13521: Implemented  OpenAPI docs for EventRestService

uguggilam opened the pull request:
### All Contributors

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/${JIRA-ISSUE-NUMBER}

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/OpenNMS/opennms?pullRequest=3611) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/OpenNMS/opennms?pullRequest=3611) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/OpenNMS/opennms?pullRequest=3611) it.</sub>

uguggilam commented:
1. Reference to https://docs.opennms.com/horizon/28.0.2/development/rest/events.html , updated EventInerfaceAPI to group GET methods with Open API.
2. Added API contract for POST to InterfaceAPI but not grouped under 'Events' as the Swagger generated model is not working.
2. Updated Swagger descriptions, API response codes and service class to add unimplemented API contracts.
3. Unit tested updated changes in local. please find attached screenshots for reference
<img width="1440" alt="Screen Shot 2021-09-21 at 12 27 11 PM" src="https://user-images.githubusercontent.com/88553957/134230266-3076db76-15af-4864-9a5b-b087ae6a1276.png">
<img width="1440" alt="Screen Shot 2021-09-21 at 12 13 46 PM" src="https://user-images.githubusercontent.com/88553957/134230270-19bdbedc-d7b5-407e-acd7-c12dcebddceb.png">

cgorantla commented:
Validated these APIs, everything is working as before.
